### DISCUSSION TITLE
Partial packet instances

### DISCUSSION THREAD

**Discussion by JackNWhite** [nasa/fprime#2319] [Ideas]
Are there any plans to allow one to specify in FPP partial instantiation of packets?

A rather contrived, but simple, example:
```
enum MySystemErrorClass {
  CASCADING
  ISOLATED
  USER
}

event MySystemError(
  class:        MySystemErrorClass,
  affects:      HwSubsystemBitField,
  internal_err: MySystemResult
) \
  severity warning high \
  format "System error / Cls: {} Affects: {} Err code: {}"

event CascadingSystemError is MySystemError(class=MySystemErrorClass.CASCADING) \
  format "Cascading system error / Affects: {} Error code: {}"
event IsolatedSystemError  is MySystemError(class=MySystemErrorClass.ISOLATED) \
  format "Isolated system error / Affects: {} Error code: {}"
event UserSystemError      is MySystemError(class=MySystemErrorClass.USER)  \
  format "User system error / Affects: {} Error code: {}"
```

A circumstance in which I would actually like to use such a system is when I would like to expand the effective event ID space by having a second ID field within the event payload, but would prefer to have a good explanatory message in the text event log, based on the extended ID, rather than an enum value you have to look up.

**Comment by Joshua-Anderson** [comment]
What is your motivator for subclassing EVRs? Are you hitting opcode identifier space limits? Are you trying to save bytes by going down to a smaller opcode size and adding an extra identifier byte only when needed?

**Reply by JackNWhite** [reply]
We have implemented Space Packet in the framing protocol, using a secondary header with a 1-byte secondary packet ID. This is shared with another computer, not running F'. This is mostly just fine, but in one of our subsystems, we have lots of error events which are the same, but may be raised in many different places. There is a tertiary ID which specifies the location at which the error arose. It is cumbersome and mistake-prone to fully define a lot of events in which one field is fixed.

**Comment by LeStarch** [comment]
@bocchino FPP language suggestion
